src/classes/frModel.cls

```diff
@@ -56,32 +72,84 @@ public with sharing abstract class frModel {
     }
     
     protected void applyMappings(SObject record, Map<String, Object> request) {
-    	Map<String, Schema.SObjectField> fields = frSchemaUtil.getFields(record.getSObjectType().getDescribe().getName());
-        String funraiseId = String.valueOf(request.get('id'));
+    	String sObjectName = record.getSObjectType().getDescribe().getName();
+        Map<String, Schema.SObjectField> fields = frSchemaUtil.getFields(sObjectName);
         
         //a single funraise field might flow into multiple salesforce fields
-        Map<String, List<frMapping__c>> frNameToSfName = new Map<String, List<frMapping__c>>();
+        Map<String, List<frMapping__c>> frFieldToMappings = new Map<String, List<frMapping__c>>();
         List<frMapping__c> constantMappings = new List<frMapping__c>();
+        Set<String> fieldsToQueryFor = new Set<String>();
+        Boolean needLastModifiedDate = false;
+        
         for(frMapping__c mapping : getMappings()) {
             if(mapping.Is_Constant__c) {
                 constantMappings.add(mapping);
             } else {
                 List<frMapping__c> mappings;
-                if(frNameToSfName.containsKey(mapping.fr_Name__c)) {
-                    mappings = frNameToSfName.get(mapping.fr_Name__c);
+                if(frFieldToMappings.containsKey(mapping.fr_Name__c)) {
+                    mappings = frFieldToMappings.get(mapping.fr_Name__c);
                 } else {
                     mappings = new List<frMapping__c>();
                 }
                 mappings.add(mapping);
-                frNameToSfName.put(mapping.fr_Name__c, mappings);
+                frFieldToMappings.put(mapping.fr_Name__c, mappings);
+            }
+            
+            if(mapping.Conflict_Resolution__c == MAPPING_NO_OVERWRITE) {
+                //We need to know the existing values of these fields
+                //if null, we can sync the incoming data.  Otherwise, we don't overwrite
+                fieldsToQueryFor.add(mapping.sf_Name__c);
             }
+            if(!needLastModifiedDate && mapping.Conflict_Resolution__c == MAPPING_OVERWRITE_RECENT) {
+                needLastModifiedDate = true;
+            }
+        }        
+        if(needLastModifiedDate) {
+        	fieldsToQueryFor.add('LastModifiedDate');    
         }
-
-        for(String fieldName : frNameToSfName.keySet()) {
+        
+        String funraiseId = getFunraiseId();
+        String salesforceId = getSalesforceId();
+        SObject queriedFieldsRecord = null;
+        if(!fieldsToQueryFor.isEmpty()) {
+            String query = 'SELECT ' + String.join(new List<String>(fieldsToQueryFor), ',') 
```

Comment:
I'm not super familiar with the syncs, do we ever sync more than one? Do we need to worry about bulkifying this?

Reply:
Currently, no.  It's one mappable record per sync/lifecycle.   And I think even if we did I would imagine this would mostly stay the same, but we would update the query to be `WHERE Id IN (list of things we're syncing)` since the columns in this query are pretty static for every incoming sync based on how they setup their mappings